File: LibraryManagementSystem/Backend/LibraryManagementSystem.Entities/Models/BookLoan.cs
namespace LibraryManagementSystem.Entities.Models
{
    public class BookLoan
    {
        public int BookLoanId { get; set; }
        public int StudentId { get; set; }
        public int BookId { get; set; }
        public string? TypeOfLoan { get; set; }
        public string? State { get; set; }
        public DateTime LoanDate { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime ReturnDate { get; set; }
    }
}

File: LibraryManagementSystem/Frontend/library-management-system-web/src/app/entities/models/library/book-loan.ts
export class BookLoan {
    bookLoanId: number | undefined;
    studentId: number | undefined;
    bookId: number | undefined;
    typeOfLoan: string | null | undefined;
    state: string | null | undefined;
    loanDate: Date | undefined;
    dueDate: Date | undefined;
    returnDate: Date | undefined;
}
